io-sim/test/Test/STM.hs

```diff
@@ -739,6 +771,12 @@ shrinkExpr (ExprName (Name _ (TyRepVar _))) = []
 freeNamesTerm :: Term t -> Set NameId
 freeNamesTerm (Return e)      = freeNamesExpr e
 freeNamesTerm (Throw  e)      = freeNamesExpr e
+-- A catch handler should actually have an argument, and then the implementation
+-- should handle it. But since current implementation of catch never binds the
```

Comment:
This sentence scared me! `But since current implementation of catch never binds the variable`

The implementation of `catch` does that, but the generator used in this test only ever generates handlers that ignore the variable.

So let's reword this sentence to clarify that it's the generation of the handler argument that is degenerate, not _the implementation_ 👍

Reply:
Added a description about limitation of catch term generator.